tests/destinations/test_path_utils.py

```diff
@@ -51,19 +127,168 @@ def test_get_table_prefix_layout() -> None:
+    path = create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        load_package_timestamp=now.to_iso8601_string(),
+        file_name=job_info.file_name(),
+    )
+
+    assert timestamp in path
+    assert path.endswith(f"{timestamp}.{job_info.file_format}")
+
+
+def test_create_path_resolves_current_datetime(test_load: TestLoad) -> None:
+    """Check the flow when the current_datetime is passed
+
+    Happy path checks
+
+    1. Callback,
+    2. pendulum.DateTime
+
+    Failures when neither callback nor the value is not of pendulum.DateTime
+    """
+    load_id, job_info = test_load
+    now = pendulum.now()
+    now_timestamp = now.to_iso8601_string()
+    calls = 0
+
+    def current_datetime_callback():
+        nonlocal calls
+        calls += 1
+        return now
+
+    create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        current_datetime=current_datetime_callback,
+        load_package_timestamp=now_timestamp,
+        file_name=job_info.file_name(),
+    )
+
+    create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        current_datetime=now,
+        load_package_timestamp=now_timestamp,
+        file_name=job_info.file_name(),
+    )
+
+    # expect only one call
+    assert calls == 1
+
+    # If the value for current_datetime is not pendulum.DateTime
+    # it should fail with RuntimeError exception
+    with pytest.raises(AttributeError):
+        create_path(
+            "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+            schema_name="schema_name",
+            load_id=load_id,
+            current_datetime="now",  # type: ignore
+            load_package_timestamp=now_timestamp,
+            file_name=job_info.file_name(),
+        )
+    with pytest.raises(RuntimeError):
+        create_path(
+            "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+            schema_name="schema_name",
+            load_id=load_id,
+            current_datetime=lambda: 1234,  # type: ignore
+            load_package_timestamp=now_timestamp,
+            file_name=job_info.file_name(),
+        )
+
+
+def test_create_path_uses_current_moment_if_current_datetime_is_not_given(
+    test_load: TestLoad, mocker
+) -> None:
+    load_id, job_info = test_load
+    logger_spy = mocker.spy(logger, "info")
+    pendulum_spy = mocker.spy(pendulum, "now")
+    create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        file_name=job_info.file_name(),
+    )
+    logger_spy.assert_called_once_with("current_datetime is not set, using pendulum.now()")
+    pendulum_spy.assert_called()
```

Comment:
heh cool but also check path because this is what you are testing

Reply:
Done.